virtool/indexes/api.py

```diff
@@ -349,6 +351,61 @@ async def upload(req):
     return json_response(index_file, headers=headers, status=201)
 
 
+@routes.jobs_api.patch("/api/indexes/{index_id}")
+async def finalize(req):
+    """
+    Finalize an index by setting `ready` to `True` and update its `last_index_version` field.
+
+    """
+    db = req.app["db"]
+    pg = req.app["pg"]
+
+    index_id = req.match_info["index_id"]
+
+    index = await db.indexes.find_one(index_id)
+
+    if index is None:
+        return not_found("Index does not exist")
+
+    try:
+        ref_id = index["reference"]["id"]
+    except KeyError:
+        return bad_request("Index is not associated with a reference")
+
+    reference = await db.references.find_one(ref_id)
+
+    if reference is None:
+        return not_found("Reference associated with index does not exist")
+
+    if not await db.otus.find_one({"reference.id": ref_id}):
+        return not_found("OTU associated with reference does not exist")
+
+    async with AsyncSession(pg) as session:
+        query = await session.execute(
+            select(IndexFile).filter_by(reference=ref_id))
```

Comment:
Turn this into a reusable tested function called `get_index_files(index_id)`. Will be used in ch-641.